llvm/lib/CASObjectFormats/NestedV1.cpp

```diff
@@ -1853,9 +1857,14 @@ Error LinkGraphBuilder::makeSymbol(Expected<SymbolRef> Symbol) {
   //
   // FIXME: Handle abstract backedges somehow more directly (likely as part of
   // the above).
-  while (!Worklist.empty())
-    if (Error E = defineSymbol(Worklist.pop_back_val()))
+  while (!Worklist.empty()) {
+    SymbolToDefine Sym = Worklist.pop_back_val();
+    // Insert sub-symbols in reverse order (from same insertion point) so when
+    // they get popped they will get processed in original order.
+    SaveAndRestore<size_t> SAR(WLInsertionPoint, Worklist.size());
+    if (Error E = defineSymbol(Sym))
       return E;
+  }
```

Comment:
`defineSymbol` will only add direct children to the work list. I think you don't need `SaveAndRestore` (which I didn't know about BTW... nice utility). I believe reversing the added nodes has the same effect as the code you wrote, but without doing insertions mid-vector:
```
size_t OldSize = Worklist.size();
if (Error E = defineSymbol(Sym))
  return E;

// Reverse the order of any children added to keep them in edge order.
size_t NumAdded = Worklist.size() - OldSize;
if (NumAdded > 2)
  llvm::reverse(makeMutableArrayRef(Worklist).take_back(NumAdded));
```

Reply:
Should be `NumAdded > 1`. Could also just always call `llvm::reverse` to protect against bugs like that :). And the comment should be "any children to visit the earliest edge first".

Reply:
Ah, and `llvm::reverse` is the wrong thing; it doesn't wrap `std::reverse` (https://en.cppreference.com/w/cpp/algorithm/reverse), as I thought it did. Instead you want:
```
std::reverse(Worklist.begin() + OldSize, Worklist.end());
```

Reply:
> `defineSymbol` will only add direct children to the work list.

Ah, I hadn't realized that, I was under the impression that `defineSymbol` could be called recursively. I'll adopt your `reverse` suggestion.